scripts/partition_manager.py

```diff
@@ -416,6 +416,29 @@ def first_partition_has_been_aligned(first, solution):
 
 
 def _set_addresses_and_align(reqs, sub_partitions, solution, size, start, dynamic_partitions, dp):
+    # Resolve 'align_next'
+    solution_non_empty = list(filter(lambda part: "EMPTY_" not in part, solution))
+    for i in range(len(solution_non_empty) - 1):
+        if 'placement' in reqs[solution_non_empty[i]] and 'align_next' in reqs[solution_non_empty[i]]['placement']:
+            alignment = reqs[solution_non_empty[i]]['placement']['align_next']
+            if 'placement' not in reqs[solution_non_empty[i + 1]]:
+                reqs[solution_non_empty[i + 1]]['placement'] = dict()
+            if 'align' not in reqs[solution_non_empty[i + 1]]['placement']:
+                reqs[solution_non_empty[i + 1]]['placement']['align'] = dict()
+            else:
+                if 'start' in reqs[solution_non_empty[i + 1]]['placement']['align']:
+                    a_start = reqs[solution_non_empty[i + 1]]['placement']['align']['start']
+                    a_next = reqs[solution_non_empty[i]]['placement']['align_next']
+                    if ((a_start % a_next) == 0) or ((a_next % a_start) == 0):
+                        alignment = max(a_next, a_start)
+                    else:
+                        raise PartitionError(f"Align config already present in {solution_non_empty[i + 1]}")
+                else:
+                    raise PartitionError(f"Align config already present in {solution_non_empty[i + 1]}")
+
+            reqs[solution_non_empty[i + 1]]['placement']['align']['start'] = alignment
+            del reqs[solution_non_empty[i]]['placement']['align_next']
```

Comment:
If we no longer want `align_next` in the dict, why not call `reqs[solution_non_empty[i]]['placement'].pop('align_next')`? Using `del` here makes me think that the variable needs to be obliterated each loop for memory reasons.

Don't know which is more efficient to use, especially considering the number of times this is called via loop cycles * function calls. Would be interesting to see for academic reasons if this is being called _a lot_. Feel free to ignore though :)

Reply:
I ended up with a pop at the start instead.